test/OperationTest.cpp

```diff
@@ -489,36 +492,32 @@ TEST(Operation, verifyLimitIsProperlyAppliedAndUpdatesRuntimeInfoCorrectly) {
   auto result = valuesForTesting.runComputation(
       timer, ComputationMode::LAZY_IF_SUPPORTED);
 
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numCols_, 0);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numRows_, 0);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().children_.at(0)->numCols_, 0);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().children_.at(0)->numRows_, 0);
-
-  auto& idTables = result.idTables();
-
-  auto iterator = idTables.begin();
-  ASSERT_NE(iterator, idTables.end());
-
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numCols_, 2);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numRows_, 0);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().children_.at(0)->numCols_, 2);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().children_.at(0)->numRows_, 1);
-
-  ++iterator;
-  ASSERT_NE(iterator, idTables.end());
-
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numCols_, 2);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numRows_, 1);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().children_.at(0)->numCols_, 2);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().children_.at(0)->numRows_, 3);
-
-  ++iterator;
-  ASSERT_EQ(iterator, idTables.end());
+  auto& rti = valuesForTesting.runtimeInfo();
+  auto& childRti = *rti.children_.at(0);
 
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numCols_, 2);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().numRows_, 1);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().children_.at(0)->numCols_, 2);
-  EXPECT_EQ(valuesForTesting.runtimeInfo().children_.at(0)->numRows_, 3);
+  EXPECT_EQ(rti.numCols_, 0);
+  EXPECT_EQ(rti.numRows_, 0);
+  EXPECT_EQ(childRti.numCols_, 0);
+  EXPECT_EQ(childRti.numRows_, 0);
+
+  expectAtEachStageOfGenerator(std::move(result.idTables()),
+                               {[&]() {
+                                  EXPECT_EQ(rti.numCols_, 2);
+                                  EXPECT_EQ(rti.numRows_, 0);
+                                  EXPECT_EQ(childRti.numCols_, 2);
+                                  EXPECT_EQ(childRti.numRows_, 1);
+                                },
+                                [&]() {
+                                  EXPECT_EQ(rti.numCols_, 2);
+                                  EXPECT_EQ(rti.numRows_, 1);
+                                  EXPECT_EQ(childRti.numCols_, 2);
+                                  EXPECT_EQ(childRti.numRows_, 3);
+                                }});
+
+  EXPECT_EQ(rti.numCols_, 2);
+  EXPECT_EQ(rti.numRows_, 1);
+  EXPECT_EQ(childRti.numCols_, 2);
+  EXPECT_EQ(childRti.numRows_, 3);
```

Comment:
Also with a generator lambda that abstracts away these four lines, then I am happy here:) (again 4 occasions).